### PR TITLE
Auto generate Volume Name if not specified in Request

### PR DESCRIPTION
**Pull request by aravindavk**
Updates: #851
Signed-off-by: Aravinda VK <avishwan@redhat.com>

### CODE REVIEW

**Comment by Madhu-1** [comment]
@aravindavk is it possible to add a test case for this.

File: glustercli/cmd/volume-create.go [hunk 1]
```diff
@@ -45,12 +46,13 @@ var (
 		Use:   "create <volname> [<brick> [<brick>]...|--size <size>]",
```
Comment (Madhu-1) [inline glustercli/cmd/volume-create.go]: need to change this use

Comment (aravindavk) [reply glustercli/cmd/volume-create.go]: updated. Thanks

**Comment by aravindavk** [comment]
> LGTM, only concern is in case of volume name is empty we are logging it in CLI.

That should be fine I guess, it will reflect the user input.

**Comment by aravindavk** [comment]
> @aravindavk not related to this. Just a question
we have Volume struct and VolCreateReq struct in req_resp.go and VolCreateReq struct in volume_req.go. are we using all these structures for volume create?

We are using all of the struct, but we need to unify these structs. Once we decide on API endpoint for smartvol and regular volume create, we can start unifying them

File: glustercli/cmd/volume-create.go [hunk 2]
```diff
@@ -133,17 +133,16 @@ func volumeCreateCmdRun(cmd *cobra.Command, args []string) {
 		return
 	}
 
-	if len(args) < 2 {
+	if len(args) < 1 {
```
Comment (raghavendra-talur) [inline glustercli/cmd/volume-create.go]: what happens when name is specified but bricks are omitted?

Comment (aravindavk) [reply glustercli/cmd/volume-create.go]: Converted name as flag(`--name=<name>`). args will contain bricks list only.

**Comment by aravindavk** [comment]
I realized that, I need to update documentation and args upgrade script. Do not merge.

I will also add test case as @Madhu-1 suggested

File: glustercli/cmd/volume-create.go (hunk 2 above)
Comment (raghavendra-talur) [reply glustercli/cmd/volume-create.go]: thanks!

**Comment by aravindavk** [comment]
Added tests and updated documentation. Please review

**Pull request merged**